### PR TITLE
Update changelog for new version 0.5

### PR DESCRIPTION
**Pull request by McdonaldSeanp**

### CODE REVIEW

File: #CHANGELOG.md# [hunk 1]
```diff
@@ -0,0 +1,150 @@
+# Change Log
+All notable changes to this project will be documented in this file.
+This project adheres to [Semantic Versioning](http://semver.org/).
+
+This changelog adheres to [Keep a CHANGELOG](http://keepachangelog.com/).
+
+## [Unreleased]
+## [0.4.1] - 2015-11-12
+### Changed
+- Fixed problem when tools use out of tree builds by adding a build_d
```
Comment (mckern) [inline #CHANGELOG.md#]: Tell me more about your `build_d`

Comment (McdonaldSeanp) [inline #CHANGELOG.md#]: Too late, I'm too fast for you

**Comment by McdonaldSeanp** [comment]
This isn't done yet

File: CHANGELOG.md [hunk 2]
```diff
@@ -5,11 +5,19 @@ This project adheres to [Semantic Versioning](http://semver.org/).
...
 
 ## [Unreleased]
+## [0.4.2] - 2015-12-22
+### Fixed
+- Fixed problem when tools use out of tree builds
+- Vanagon no longer changes group ownership on
+  Solaris 10 installs
+
+### Changed
+- Added lock_manager gem to vanagon project
```
Comment (stahnma) [inline CHANGELOG.md]: Added Hardware engine is probably the more important part of that statement to vanagon users.

Comment (McdonaldSeanp) [inline CHANGELOG.md]: Oh, whoops, yeah left that out

Comment (mckern) [inline CHANGELOG.md]: This is more of an `Added` and less of a `Changed`

**Comment by stahnma** [comment]
This still a don't merge?

**Comment by McdonaldSeanp** [comment]
I think so, I can't do one of these : [0.4.1]: https://github.com/puppetlabs/vanagon/compare/0.4.0...0.4.1 until I've pushed a new tag which I can't do without a key

File: CHANGELOG.md [hunk 3]
```diff
@@ -5,11 +5,25 @@ This project adheres to [Semantic Versioning](http://semver.org/).
 This changelog adheres to [Keep a CHANGELOG](http://keepachangelog.com/).
 
 ## [Unreleased]
+## [0.5.0] - 2015-12-22
+### Changed
+- updated the way makefiles are generated
```
Comment (mckern) [inline CHANGELOG.md]: I would probably write this out a little more. I'll give you some copy to think about in a minute.

Comment (mckern) [inline CHANGELOG.md]:
```
- Replaced the ERB template based Makefile generation with a procedural Makefile backend.
```

Comment (mckern) [inline CHANGELOG.md]: I'm not wedded to my copy; we could probably pull verbiage from @adrienthebo's PR and commits on this. But it's probably the single biggest backend change and the one most likely to expose edge cases, so I'd want to make sure it's called out/spelled out for users.

File: CHANGELOG.md [hunk 4]
```diff
@@ -5,11 +5,25 @@ This project adheres to [Semantic Versioning](http://semver.org/).
...
+
+### Fixed
+- Fixed problem when tools use out of tree builds
```
Comment (mckern) [inline CHANGELOG.md]: `out-of-tree` could stand hyphenation. This might need a little more detail, though.

File: CHANGELOG.md [hunk 5]
```diff
@@ -5,11 +5,25 @@ This project adheres to [Semantic Versioning](http://semver.org/).
...
+- Vanagon no longer changes group ownership on
+  Solaris 10 installs
+
+### Added
+- Added lock_manager gem to vanagon project
+- Added hardware engine
+- Added support for .vim .json and .service config files
```
Comment (mckern) [inline CHANGELOG.md]: wxnvc

File: CHANGELOG.md [hunk 6]
```diff
@@ -5,11 +5,26 @@ This project adheres to [Semantic Versioning](http://semver.org/).
 This changelog adheres to [Keep a CHANGELOG](http://keepachangelog.com/).
 
 ## [Unreleased]
+## [0.5.0] - 2015-12-22
+### Changed
+- Replaced the ERB template based Makefile generation with a procedural Makefile backend
```
Comment (mckern) [inline CHANGELOG.md]: `ERB template-based`? To hyphenate or not to hyphenate, that is **a** question.

File: CHANGELOG.md [hunk 7]
```diff
@@ -5,11 +5,26 @@ This project adheres to [Semantic Versioning](http://semver.org/).
...
+
+### Fixed
+- Added build_dir function for when tools need to move away from the regular build
```
Comment (mckern) [inline CHANGELOG.md]: `need to move away` or `need to be moved away`?

File: CHANGELOG.md [hunk 8]
```diff
@@ -5,11 +5,26 @@ This project adheres to [Semantic Versioning](http://semver.org/).
...
+  directory to avoid conflicts with build files and source files
+- Vanagon no longer changes group ownership on
+  Solaris 10 installs
```
Comment (mckern) [inline CHANGELOG.md]:
```
Vanagon-generated packages no longer changes group 
ownership of files when installed on Solaris 10
```

File: CHANGELOG.md [hunk 9]
```diff
@@ -5,11 +5,28 @@ This project adheres to [Semantic Versioning](http://semver.org/).
...
+### Changed
+- Replaced the ERB template based Makefile generation with a procedural
+  Makefile backend
+
+### Fixed
+- Added build_dir function for when tools need to be moved away from the
+  regular build directory to avoid conflicts with build files and source
+  files
+- Vanagon no longer changes group ownership on
+  Solaris 10 installs
```
Comment (mckern) [inline CHANGELOG.md]:
```
Vanagon-generated packages no longer change group 
ownership of files when installed on Solaris 10
```

**Comment by mckern** [comment]
I'm out of nitpicks. :+1:

**Pull request merged**